src/extractcode/extract.py

```diff
@@ -151,21 +151,21 @@ def extract(location, kinds=extractcode.default_kinds, recurse=False):
             target = join(abspath(top), extractcode.get_extraction_path(loc))
             if TRACE:
                 logger.debug('extract:target: %(target)r' % locals())
-            for xevent in extract_file(loc, target, kinds):
+            for xevent in extract_file(loc, target, kinds, remove_originals=remove_originals):
                 if TRACE:
                     logger.debug('extract:walk:extraction event: %(xevent)r' % locals())
                 yield xevent
 
             if recurse:
                 if TRACE:
                     logger.debug('extract:walk: recursing on target: %(target)r' % locals())
-                for xevent in extract(target, kinds, recurse):
+                for xevent in extract(target if remove_originals else localtion, kinds, recurse, remove_originals=remove_originals):
```

Comment:
`localtion` likely would need to be `location`...

Reply:
Good catch,  i though I had fixed that.